src/main/webapp/script.js

```diff
@@ -36,4 +36,124 @@ function displayError() {
   // TODO: Write error to user interface
   console.log('Geolocation not enabled');
   alert(alertMessage);
+}
+
+//Array of the (currently 6 for this demo build) 15 listings gathered from the fetch request
+const listingsArray = [];
+let count = 0;
+
+function displayList() {
+  fetch('/data').then(response => response.json()).then((listings) => {
+    listings.forEach((listing) => {
+      listingsArray.push(createResultCard(listing.name, listing.address, listing.image, listing.rating, listing.website, count));
+      count++;
+    });
+    limitList('begin');
+  });
+}
+
+//Variable keeping track of the card that is currently shown at the bottom of the list
+let currentEndCard;
+
+/**
+ * @param {string} direction Contains which direction the list should iterate
+ */
+function limitList(direction) {
```

Comment:
If you're interested, you can move the pagination bits to a different script. It might allow for improved modularity, and parsing later on.

Reply:
Good idea, doing that now.